### PR TITLE
HW Task5 with tests 07.11.2016

### PR DESCRIPTION
**Pull request by KristinaKosko**

### CODE REVIEW

File: Task5Test/src/main/java/com/company/Main.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    sides[i] = getSides.getUserSide();
+                }
+            } catch (Exception e) {
+                System.out.println("You entered wrong side!!!");
+                System.exit(1);
+            }
+            Triangle triangle = new Triangle(sides[0], sides[1], sides[2]);
+            triangle.definesType();
+            type = triangle.getType();
+            if (type == 1) {
```
Comment (AlexanderSoroka) [inline Task5Test/src/main/java/com/company/Main.java]: Use constant instead magic numbers

File: Task5Test/src/test/java/com/company/TriangleTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.company;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class TriangleTest {
```
Comment (AlexanderSoroka) [inline Task5Test/src/test/java/com/company/TriangleTest.java:9]: Is that it? You must be joking. Please add the tests for all triangles types and positive/negative scenarious

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: Task5Test/src/test/java/com/company/TriangleTest.java [hunk 3]
```diff
@@ -1,38 +1,95 @@
...
-        triangle = new Triangle(2, 2, 4);
-        triangle.definesType();
+    @DataProvider(name = "ValidIsoscelesTriangle")
+    public Object[][] setValidSidesOfTheIsoscelesTriangle() {
+        return new Object[][] {
+                {BigDecimal.valueOf(10), BigDecimal.valueOf(10), BigDecimal.valueOf(11)},
+                {BigDecimal.valueOf(10), BigDecimal.valueOf(11), BigDecimal.valueOf(10)},
+                {BigDecimal.valueOf(11), BigDecimal.valueOf(10), BigDecimal.valueOf(10)},
+
+                {BigDecimal.valueOf(Double.MAX_VALUE), BigDecimal.valueOf(Double.MAX_VALUE), BigDecimal.valueOf(11)},
```
Comment (AlexanderSoroka) [inline Task5Test/src/test/java/com/company/TriangleTest.java]: I don't think there is a sense to test Double.MAX_VALUE if you use BigDecimal. All such values will be in different equivalence classes for the double arguments, not BigDecimal.

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

**Review by AlexanderSoroka — APPROVED** [review]
Well done!
